### PR TITLE
Provide API to detect integrated graphics

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
This PR adds a new property `usingIntegratedGraphics` to `WebGLRenderCompatibilityInfo`.

If true, there is a likelihood that integrated graphics are being used.  This property has the possibility of providing false positives and negatives. A user should use this property mainly as a hint and manually verify what graphics chip is being used.

~~TODO:~~
- ~~AMD chips?~~ It appears that AMD integrated graphics are not usually paired with a graphics card so detecting integrated usage there is less important than Intel.
- ~~Other Intel Chips aside from HD and Iris?~~ Detecting back to 2011 seems adequate.  See comment section.
- ~~What about iPhones, iPads, M1 chipsets?  Technically these are integrated - should they count as such?~~ See comment section.

### CODE REVIEW

File: core/webgl-compatibility/src/Capabilities.ts [hunk 1]
```diff
@@ -73,6 +73,24 @@ const buggyIntelMatchers = [
...
+  /Iris (Pro|Plus|Xe)/,
+];
+
+function isIntegratedGraphics(args: {unmaskedVendor?: string, unmaskedRenderer?: string}): boolean {
+  if (args.unmaskedVendor && args.unmaskedVendor.includes("Intel")) {
+    if (args.unmaskedRenderer && integratedIntelGpuMatchers.some((x) => x.test(args.unmaskedRenderer!)))
+      return true;
+  }
+
+  // ###TODO: Detect AMD integrated graphics
```
Comment (markschlosseratbentley) [inline core/webgl-compatibility/src/Capabilities.ts:89]: What about AMD integrated graphics?  A good way to detect them?

File: core/webgl-compatibility/src/Capabilities.ts [hunk 2]
```diff
@@ -73,6 +73,24 @@ const buggyIntelMatchers = [
...
+
```
Comment (markschlosseratbentley) [inline core/webgl-compatibility/src/Capabilities.ts:90]: How should we categorize Apple M1 chipsets and mobile chipsets?  Technically these are integrated as well.

**Comment by pmconne** [comment]
> iPhones, iPads, M1 chipsets? Technically these are integrated - should they count as such?

The motivation for this feature is devices that may possess both a discrete and integrated GPU. We want to detect if they are using the integrated GPU, and suggest that they switch to the discrete GPU if they have one (we can't detect that of course).
iPads and iPhones only have one GPU.

File: core/webgl-compatibility/src/Capabilities.ts (hunk 2 above)
Comment (markschlosseratbentley) [reply core/webgl-compatibility/src/Capabilities.ts:90]: See Paul's comment about the motivation being for devices possessing both discrete and integrated - so we should not count Apple M1 and mobile chipsets.

File: core/webgl-compatibility/src/Capabilities.ts [hunk 3]
```diff
@@ -73,6 +73,24 @@ const buggyIntelMatchers = [
...
+  return false;
+}
+
```
[the remark this answers is not in the document]
Comment (pmconne) [reply core/webgl-compatibility/src/Capabilities.ts:93]: Your regex looks like it works for all Intel integrated chips going back to 2011. That seems more than adequate.

File: core/webgl-compatibility/src/RenderCompatibility.ts [hunk 4]
```diff
@@ -109,6 +109,11 @@ export interface WebGLRenderCompatibilityInfo {
   unmaskedRenderer?: string;
   /** The vendor string reported by this client's graphics driver. */
   unmaskedVendor?: string;
+  /** If true, there is a likelihood that integrated graphics are being used. This can be used to warn users on systems with both integrated graphics and dedicated graphics that they should try to switch to their dedicated graphics for better performance.
+   * @note This property has the possibility of providing false positives and negatives. A user should use this property mainly as a hint and manually verify what graphics chip is being used.
+   * @beta
```
Comment (pmconne) [inline core/webgl-compatibility/src/RenderCompatibility.ts:114]: Why beta?

**Review by pmconne — CHANGES_REQUESTED** [review]
Add simple regression tests.

File: core/webgl-compatibility/src/RenderCompatibility.ts (hunk 4 above)
Comment (markschlosseratbentley) [reply core/webgl-compatibility/src/RenderCompatibility.ts:114]: You think public?  It's pretty basic API, I suppose.

Comment (pmconne) [reply core/webgl-compatibility/src/RenderCompatibility.ts:114]: Yes.

**Review by pmconne — APPROVED** [review]

**Pull request merged**